sync/handlers/handler.go

```diff
@@ -48,6 +49,9 @@ func NewSyncHandler(
 		stateTrieLeafsRequestHandler: NewLeafsRequestHandler(evmTrieDB, provider, networkCodec, stats),
 		blockRequestHandler:          NewBlockRequestHandler(provider, networkCodec, stats),
 		codeRequestHandler:           NewCodeRequestHandler(evmTrieDB.DiskDB(), networkCodec, stats),
+
+		// TODO: need to pass in the warp backend from the subnet-evm.
+		signatureRequestHandler: NewSignatureRequestHandler(nil, networkCodec, stats),
```

Comment:
we shouldn't add this to the `sync/handlers` package, may need some refactoring.

Reply:
if we pass `nil` in here, then if a peer sends a signature request, this node will panic with a nil pointer error.

Since we're merging un-activated code in, we need to gate it so that it cannot actually be hit and cause an issue. Can we create a `NoopSignatureRequestHandler` and use that here instead until we are ready to activate this?

It would also be alright to actually construct the warp backend and pass it in here since there will not be any messages in it and it will just end up dropping each request, but using a noop handler will be even better because it will not cause the request handler to hit the database.

Reply:
Agreed we should refactor this package out now that we are putting a non-sync request handler in here. Can we move this out to a top-level `handlers` package for now?

I think we may want to refactor the networking code more, but just moving it out from here is good for this PR.

Reply:
Refactored out the `SignatureRequestHandler` and created a `NetworkHandler` which should include sync and non-sync related handlers. Also made corresponding stats.